lib/ansible/modules/tacp/tacp_instance.py

```diff
@@ -82,15 +82,15 @@ def run_module():
         api_key=dict(type='str', required=True),
         name=dict(type='str', required=True),
         state=dict(type='str', default='present',
-                   choices=['present', 'absent']),
-        datacenter=dict(type='str', required=True),
-        mz=dict(type='str', required=True),
-        storage_pool=dict(type='str', required=True),
-        template=dict(type='str', required=True),
-        vcpu_cores=dict(type='int', required=True),
-        memory=dict(type='str', required=True),
-        disks=dict(type='list', required=True),
-        nics=dict(type='list', required=True),
+                   choices=['started', 'absent', 'stopped', 'restarted', 'force-restarted', 'paused', 'shutdown']),
+        datacenter=dict(type='str', required=False),
+        mz=dict(type='str', required=False),
+        storage_pool=dict(type='str', required=False),
```

Comment:
It seems like most of these fields should be required to properly bring up an instance.

Reply:
right, they would be for bringing up an instance, but for other states like power off or restart, ones where the instance already exists, those aren't necessary to identify which instance that operation would affect. a future story will be around more intense input validation, since the set of required variables varies considerably depending on what is specified, so only a couple of variables are truly REQUIRED from the get-go, without knowing what the task's intended purpose is.